### PR TITLE
Add CONTRIBUTING file per Technical Charter

### PR DESCRIPTION
**Pull request by dthaler**
Per discussion in CGC meeting today.  This PR is intended to replace PR #3436 and only one or the other of these two PRs should be merged.

Signed-off-by: Dave Thaler <dthaler@ntdev.microsoft.com>

### CODE REVIEW

**Review by achamayou — APPROVED** [review]
This is the location where GitHub expects to find this information, and Insights/Community will point here if it's committed under that name. It's where many users will expect to find it too.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Contributing to Open Enclave
+
+**Did you find a bug?**
+
+* **Do not open up a GitHub issue if the bug is a security vulnerability in Open Enclave**, and instead refer to our [security policy](docs/Community/Contributing.md#reporting-security-issues).
+
+* **Ensure the bug was not already reported** but searching on GitHub under [Issues](https://github.com/openenclave/openenclave/issues).
```
Comment (CodeMonkeyLeet) [inline CONTRIBUTING.md:7]:
```suggestion
* **Ensure the bug was not already reported** by searching on GitHub under [Issues](https://github.com/openenclave/openenclave/issues).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+Include the relevant issue number if applicable.
+
+* Before submitting, please read the
+[Contributing to Open Enclave](docs/Community/Contributing.md#general-guidelines)
+guidelines to know more about coding conventions and benchmarks.
+
+**Do you intend to add a new feature or change an existing one?**
+
+* See the
+[Contributing to Open Enclave](docs/Community/Contributing.md#general-guidelines)
```
Comment (CodeMonkeyLeet) [inline CONTRIBUTING.md:29]:
```suggestion
[Contributing to Open Enclave](docs/Community/Contributing.md#design-discussion)
```

**Comment by gupta-ak** [comment]
I'd also prefer this PR over the #3436

**Comment by dthaler** [comment]
bors try

**Comment by dthaler** [comment]
Squashed commits to make DCO happy.

bors r+

**Pull request merged**